pkg/upgrade/upgrader/helper.go

```diff
@@ -75,23 +74,17 @@ func patchDelpoyment(
 	if err != nil {
 		return err
 	}
-
-	err = retry.
-		Times(60).
-		Wait(5 * time.Second).
-		Try(func(attempt uint) error {
-			rolloutStatus, err1 := deployClient.WithNamespace(namespace).
-				RolloutStatus(deployName)
-			if err1 != nil {
-				return err1
-			}
-			if !rolloutStatus.IsRolledout {
-				return errors.Errorf("failed to rollout because %s", rolloutStatus.Message)
-			}
-			return nil
-		})
-	if err != nil {
-		return err
+	for {
+		rolloutStatus, err1 := deployClient.WithNamespace(namespace).
+			RolloutStatus(deployName)
+		if err1 != nil {
+			return errors.Wrapf(err1, "failed to get rollout status for deployment %s", deployName)
+		}
+		if rolloutStatus.IsRolledout {
+			break
+		}
+		klog.Infof("Deployment %s not yet rolled out because %s", deployName, rolloutStatus.Message)
+		time.Sleep(10 * time.Second)
```

Comment:
Can you take an ENV variable for MAX_RETRIES - default can be 10 mins. The forever loop is broken after timeout is reached. ENV, so that users can change to desired values.

Reply:
If we set a default and take env to break out of the loop. The job will crash and restart and again stuck in the same loop. This crash is not required and the deployment rollout can be delayed due to several reasons and the env may not be set in many cases. Unless the deployment is rolled out the upgrade can not proceed also. IMO having an ENV will not help us here.

Reply:
We also put up the rollout message for each attempt for debugging purposes.

Reply:
what happens if the rollout status never gets updated, how does user recover from it?

Reply:
Then the user has to check the deployment for the reason why the deployment is stuck in that state. The rollout will only fail for reasons like image pull errors, node unavailable, etc.

Reply:
Can yiou please update the testing section by manually covering a case that hits this error case?

Reply:
Done